text/0000-adjust-default-object-bounds.md

```diff
@@ -0,0 +1,275 @@
+  if that is the necessary bound, or a looser bound if that would be helpful.)
+- *If `SomeType` contains multiple where-clauses of the form `T:'a`,
+  then the default is cleared and explicit lifetiem bounds are
+  required. There are no known examples of this in the standard
+  library as this situation arises rarely in practice.*
+  
+## Legacy annotation for opting out
+
+RFC 1122 suggested that, at times, it would be appropriate to include
+an *opt-out* provision that allows users to restore older behavior, to
+ease transition to the new rules. An opt-out provision makes it easy
+to restore a crate to working order quickly.
+
+This RFC proposes an attribute `#[legacy(...)]`. This attribute can be
+placed at any place in the tree and with any arguments -- however, if
+it is placed in an unexpected location, or with unexpected arguments,
+a lint warning (unused attribute) will be signaled. It is important
+that we never signal errors so as to allow new legacy opt-outs to be
+added in the future without causing older compilers to break
+(unfortunately, the 1.0 compiler will still reject the legacy
+annotation, but adding this change now lays the groundwork for the
+future).
+
+With regard to this specific change, the annotation
+`#![legacy(default_object_bounds)]` will be accepted at the crate
+level (and only at the crate level). It will cause the older rules to
+be applied. This legacy opt-out is not time-limited and will be
+available indefinitely (this poses very little maintenance burden in
+the compiler). Any other use will yield a lint error.
+  
+# Timing and breaking change implications
+
+This is a breaking change, and hence it behooves us to evaluate the
+impact and describe a procedure for making the change as painless as
+possible. One nice propery of this change is that it only affects
+*defaults*, which means that it is always possible to write code that
+compiles both before and after the change by avoiding defaults in
+those cases where the new and old compiler disagree.
+
+The estimated impact of this change is low, for two reasons:
+- A recent test of crates.io found [no regressions][crater] caused by
+  this change (however, a [previous run] (from before Rust 1.0) found 8
+  regressions).
+- This feature was only recently stabilized as part of Rust 1.0 (and
+  was only added towards the end of the release cycle), so there
+  hasn't been time for a large body of dependent code to arise
+  outside of crates.io.
+  
+Unfortunately, the error messages produced as a result of this change
+are not especially clear. This is because they are produced through
+lifetime inference, which is one of the trickier parts of the compiler
+when it comes to error messages. The author (nmatsakis) plans to
+investigate possible changes to the compiler to improve the situation.
+
+In order to ensure that we break as little code as possible, **these
+changes would be backported to the 1.1 release.**
```

Comment:
Well, maybe. It may actually have much *bigger* impact overall, since there are probably a fair number of people who will now see a warning when in fact the code would have worked just fine with either default.